### ISSUE TITLE
`gapi.client.gmail.users` is misplaced

### ISSUE THREAD

**Issue by jsalvata** [DefinitelyTyped/DefinitelyTyped#30491]
If you know how to fix the issue, make a pull request instead.

I know the fix the issue itself, but not how to update & run tests, etc... so just providing a bug report here.

- [x] I tried using the `@types/gapi.client.gmail` package and had problems.
- [ ] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @Bolisov

Trying to compile this code:

```
gapi.client.gmail.users.messages.send({ 'userId': 'me', 'resource': { 'raw': Base64.encodeURI(rawMessage) } })
```

throws:
```
ERROR in ...: error TS2339: Property 'gmail' does not exist on type 'typeof client'.
```

This code compiles...

```
gapi.client.users.messages.send({ 'userId': 'me', 'resource': { 'raw': Base64.encodeURI(rawMessage) } })
```

But fails because gapi.client.users does not exist, even after loading `gapi.client` and `gapi.client.gmail`.

The solution is to move the `const users` definition inside the `gmail` namespace so the types declarations match the real google-provided objects.

**Comment by gwenzek** [comment]
I have the same issue.
The types for gapi don't match the actual json object.

**Comment by lucasriondel** [comment]
+1, same issue here. @Bolisov , could you check this out please ?

**Comment by lucka-me** [comment]
Fixed in #45380.
The maintained fork [Maxim-Mazurok/google-api-typings-generator](https://github.com/Maxim-Mazurok/google-api-typings-generator) could generate the correct code.

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
